src/franka_plan_runner.cc

```diff
@@ -267,34 +255,38 @@ int FrankaPlanRunner::RunFranka() {
 
         // prevent the plan from being started if robot is not running...
         if (comm_interface_->HasNewPlan() && status_ == RobotStatus::Running) {
-          dexai::log()->info("RunFranka: Got a new plan, attaching callback!");
+          dexai::log()->info(
+              "RunFranka: found a new plan, attaching callback...");
           status_has_changed = true;
-          // joint_position_callback_ or impedance_control_callback_ can be used
-          // here:
-          robot.control(joint_position_callback_);
+          // Use either joint position or impedance control callback here
+          robot.control(std::bind(&FrankaPlanRunner::JointPositionCallback,
+                                  this, std::placeholders::_1,
+                                  std::placeholders::_2));
         } else {
           if (comm_interface_->CancelPlanRequested()) {
-            log()->error("Cancel plan requested with no active plan!");
+            log()->error(
+                "RunFranka: plan cancellation requested but there is no active "
+                "plan");
             comm_interface_->ClearCancelPlanRequest();
           }
           // no plan available or paused
-          // print out status after (lcm_publish_rate_ * 40) times:
-          if (status_has_changed) {
+          // print out status if there's any change or every 10 sec
```

Comment:
Why print status every 10sec? I would prefer to only print if state changes to avoid cluttering logs

Reply:
because messages come async and the driver status is buried in the logs above the last line. every 10 sec won't clutter the logs.